hpy/tools/autogen/public_api.h

```diff
@@ -616,6 +616,24 @@ HPy HPyGlobal_Load(HPyContext *ctx, HPyGlobal global);
 HPy_ID(228)
 void _HPy_Dump(HPyContext *ctx, HPy h);
 
+/**
+ * Set the vectorcall function for the given object.
+ *
+ * In order to use the more efficient vectorcall API in HPy, you need to
+ * define slot 'HPy_tp_vectorcall_default'. This slot enables the vectorcall
+ * capability for the type and specifies the default vectorcall function that
+ * will be used for every instance of this HPy type. This should account for
+ * the most common case (every instance of an object uses the same vectorcall
+ * function) but to still provide the necessary flexibility, function
+ * 'HPyVectorcall_Set' allows to set different maybe specialized vectorcall
+ * functions for each instance. This may be done at any time but is most
+ * commonly used in the constructor of an object.
+ *
```

Comment:
I've documented the restriction that `HPy_SetCallFunction` must be done in the object constructor but I did not implement a debug mode feature that enforces it. I assume that it is possible but with significant effort since we need to keep track of various things and need to intercept calls etc etc.